File: data_gen.m
phi = 12 * pi * [0 : 999] / 1000;
sinus = sin(phi);

plot_data.data.phi.values = phi;
plot_data.data.phi.name = 'phi';
plot_data.data.phi.unit = 'rad';

plot_data.data.sinus.values = sinus;
plot_data.data.sinus.name = 'sinus';
plot_data.data.sinus.unit = 'm';

plot_data.meta.timestamp_last_sample = 123;
plot_data.meta.location = 'Deutschland, Oberkochen';
plot_data.meta.machine = 'some machine';
plot_data.meta.worker = 'Max Mustermann';

plot_data.plot.trig.figure{1}.subplot{1}.plots = {{'phi', 'sinus'}};

save('-v6', 'data.mat', 'plot_data');

File: data_gen.py
import numpy as np
from scipy.io import savemat

phi = np.linspace(0.1, 37.1, 1000)
sinus = np.sin(phi)

dic = {'plot_data': {'data': {'phi': {'values': phi, 'name': 'x_data', 'unit': 'rad'},
                              'sinus': {'values': sinus, 'name': 'y_data', 'unit': 'm'}},
                     'meta': {'timestamp_last_sample': 123,
                              'location': 'Deutschland, Oberkochen',
                              'machine': 'some machine',
                              'worker': 'Max Mustermann'},
                     'plot': {'trig': {'figure': [{'subplot': [{'plots': [['phi', 'sinus']]}]}]}}}}

savemat('data.mat', dic)
